libp2p/protocols/kademlia/types.nim

```diff
@@ -161,6 +167,21 @@ type
     buckets*: seq[Bucket]
     config*: RoutingTableConfig
 
+  Provider* = Peer
+
+  ProviderRecord* = object
+    provider*: Provider
+    expiresAt*: chronos.Moment
+    key*: Cid
+
+  ProviderManager* = ref object
+    records*: HeapQueue[ProviderRecord]
+    knownKeys*: Table[Cid, HashSet[Provider]]
+    providedKeys*: HashSet[Cid]
```

Comment:
Maybe we should call this knownCids / providedCids since the key for both the table and hashset is a Cid